lib/core/bench/db-bench.hs

```diff
@@ -359,31 +359,26 @@ bgroupWriteSeqState db = bgroup "SeqState"
             pure cps
         cps :: [WalletBench]
         cps =
-            [ snd $ initWallet (withMovingSlot i block0) $
-                SeqState
-                    (mkIntPool a i)
-                    (mkExtPool a i)
-                    emptyPendingIxs
-                    rewardAccount
-                    defaultPrefix
+            [ snd $ initWallet (withMovingSlot i block0) $ mkSeqState a i
             | i <- [1..n]
             ]
 
 benchPutSeqState :: DBLayerBench -> [WalletBench] -> IO ()
 benchPutSeqState DBLayer{..} cps = do
     unsafeRunExceptT $ mapExceptT atomically $ mapM_ (putCheckpoint testWid) cps
 
-mkExtPool :: Int -> Int -> AddressPool  'UtxoExternal ShelleyKey
-mkExtPool numAddrs i =
-    mkAddressPool @'Mainnet (ParentContextUtxo ourAccount) defaultAddressPoolGap addrs
-  where
-    addrs = [ force (mkAddress i j, Unused) | j <- [1..numAddrs] ]
-
-mkIntPool :: Int -> Int -> AddressPool 'UtxoInternal ShelleyKey
-mkIntPool numAddrs i =
-    mkAddressPool @'Mainnet (ParentContextUtxo ourAccount) defaultAddressPoolGap addrs
+mkSeqState :: Int -> Int -> SeqState 'Mainnet ShelleyKey
+mkSeqState numAddrs _ = s 
+    { internalPool = fillPool (internalPool s)
+    , externalPool = fillPool (externalPool s)
+    }
   where
-    addrs = [ force (mkAddress i j, Unused) | j <- [1..numAddrs] ]
+    s = mkSeqStateFromAccountXPub @'Mainnet
+        ourAccount purposeCIP1852 defaultAddressPoolGap
+    fillPool (SeqAddressPool pool0) = SeqAddressPool $
+        foldl' (\p ix -> AddressPool.update (gen ix) p) pool0 [1..numAddrs]
```

Comment:
shouldn't be [1..numAddrs - 1] ? we have the first I bet in pool0 ? :thinking:

Reply:
I think that you're right — `mkSeqStateFromAccountXPub` creates two pools with `defaultAddressPoolGap` many addresses already in them, and they start from index `0`, not `1`. So, `[0…(numAddrs-1)]` would be correct (in the case where `numAddrs` is larger than the pool gap). 🤔 Then again, this is the benchmarking module, which is somewhat obsolete due to caching being on by default now; we will need to revisit this properly anyway. But there is no harm in keeping the module reasonably tidy. 🙂